src/core/services/shops.service.ts

```diff
@@ -1,67 +1,134 @@
-import { Injectable } from '@angular/core';
-import { HttpClient, HttpHeaders } from '@angular/common/http';
-import { LatLng, UserLogin, Category } from '../models/Models';
-import { Observable } from 'rxjs';
-import { Shop } from '../models/shop.model';
-import { config } from '../../configs/baseconfig'
+import {Injectable} from '@angular/core';
+import {HttpClient, HttpHeaders} from '@angular/common/http';
+import {LatLng, UserLogin, Category} from '../models/Models';
+import {Observable} from 'rxjs';
+import {Shop} from '../models/shop.model';
+import {config} from '../../configs/baseconfig';
 
 @Injectable({
-  providedIn: 'root'
+  providedIn: 'root',
 })
 export class ShopsService {
-
   private hyderabadLatng: LatLng = {
-    latitude: 17.3850,
-    longitude: 78.4867
-  }
+    latitude: 17.385,
+    longitude: 78.4867,
+  };
+
+  constructor(private http: HttpClient) {}
 
-  constructor(private http: HttpClient) { }
+  addShop(shopObject): Observable<any> {
+    const userDetails: any = JSON.parse(localStorage.getItem('shopsnearbyme'));
+    let body = new FormData();
+    body.set('name', shopObject.shopName);
+    body.set('category', shopObject.shopCategory);
+    body.set('description', shopObject.shopDescription);
+    body.set('phoneNumber', userDetails.phoneNumber);
+    body.set('timeFrom', shopObject.startTime);
+    body.set('timeTo', shopObject.endTime);
+    body.set('address', shopObject.shopAddress);
+    body.set('latitude', userDetails.latitude);
+    body.set('longitude', userDetails.longitude);
+    if (shopObject.imageFile) {
+      body.set('image', shopObject.imageFile);
+    }
+    let options = {
+      headers: this.setBasicAuthHeader({
+        email: userDetails.email,
+        password: userDetails.password,
+      }),
+    };
+    return this.http.post<any>(`${config.API_URL}/shops/others`, body, options);
+  }
 
+  getNearByDiscountsAndCoupons() {
+    const userDetails: any = JSON.parse(localStorage.getItem('shopsnearbyme'));
+    const latlng = this.sanitizeLatLng({
```

Comment:
Here the location data needs to come as a parameter. 
The location data that comes from the login details is not valid here, you should fetch the current location from the browser everytime.